include/zephyr/net/wifi.h

```diff
@@ -12,16 +12,210 @@
 #ifndef ZEPHYR_INCLUDE_NET_WIFI_H_
 #define ZEPHYR_INCLUDE_NET_WIFI_H_
 
+/* Not having support for legacy types is deliberate to enforce
+ * higher security.
+ */
 enum wifi_security_type {
 	WIFI_SECURITY_TYPE_NONE = 0,
 	WIFI_SECURITY_TYPE_PSK,
+	WIFI_SECURITY_TYPE_PSK_SHA256,
+	WIFI_SECURITY_TYPE_SAE,
+
+	__WIFI_SECURITY_TYPE_AFTER_LAST,
+	WIFI_SECURITY_TYPE_MAX = __WIFI_SECURITY_TYPE_AFTER_LAST - 1
+};
+
+/**
+ * wifi_security_txt - Get the security type as a text string
+ */
+static inline const char *wifi_security_txt(enum wifi_security_type security)
+{
+	switch (security) {
+	case WIFI_SECURITY_TYPE_NONE:
+		return "OPEN";
+	case WIFI_SECURITY_TYPE_PSK:
+		return "WPA2-PSK";
+	case WIFI_SECURITY_TYPE_PSK_SHA256:
+		return "WPA2-PSK-SHA256";
+	case WIFI_SECURITY_TYPE_SAE:
+		return "WPA3-SAE";
+	default:
+		return "UNKNOWN";
+	}
+}
+
+/* Management frame protection (IEEE 802.11w) options */
+enum wifi_mfp_options {
+	WIFI_MFP_DISABLE = 0,
+	WIFI_MFP_OPTIONAL,
+	WIFI_MFP_REQUIRED
 };
 
+/**
+ * wifi_mfp_txt - Get the MFP as a text string
+ */
+static inline const char *wifi_mfp_txt(enum wifi_mfp_options mfp)
+{
+	switch (mfp) {
+	case WIFI_MFP_DISABLE:
+		return "Disable";
+	case WIFI_MFP_OPTIONAL:
+		return "Optional";
+	case WIFI_MFP_REQUIRED:
+		return "Required";
+	default:
+		return "UNKNOWN";
+	}
+}
+
+enum wifi_frequency_bands {
+	WIFI_FREQ_BAND_2_4_GHZ = 0,
+	WIFI_FREQ_BAND_5_GHZ,
+	WIFI_FREQ_BAND_6_GHZ
+};
+
+/**
+ * wifi_mode_txt - Get the interface mode type as a text string
+ */
+static inline const char *wifi_band_txt(enum wifi_frequency_bands band)
+{
+	switch (band) {
+	case WIFI_FREQ_BAND_2_4_GHZ:
+		return "2.4GHz";
+	case WIFI_FREQ_BAND_5_GHZ:
+		return "5GHz";
+	case WIFI_FREQ_BAND_6_GHZ:
+		return "6GHz";
+	default:
+		return "UNKNOWN";
+	}
+}
+
 #define WIFI_SSID_MAX_LEN 32
 #define WIFI_PSK_MAX_LEN 64
 #define WIFI_MAC_ADDR_LEN 6
 
-#define WIFI_CHANNEL_MAX 14
+#define WIFI_CHANNEL_MAX 233
 #define WIFI_CHANNEL_ANY 255
 
+/* Based on https://w1.fi/wpa_supplicant/devel/defs_8h.html#a4aeb27c1e4abd046df3064ea9756f0bc */
+enum wifi_iface_state {
```

Comment:
WIFI_INACTIVE is when supplicant has no configuration, its just waiting for someone to configure. WIFI_INTERFACE_DISABLED is when underlying interface is disabled, but in Zephyr this is not applicable for us, as this is mainly for `rfkill` concept in Linux. But just for sake of uniformity added this.

> Another question: we have the WIFI_ASSOCIATING and WIFI_ASSOCIATED, why this flow does not apply to the authenticating and we only have the WIFI_AUTHENTICATING? 

This is because AUTHENTICATION is an intermediate state, and ASSOCIATION can be intermediate/final state depending on secure/open association.

>  I assume that for scanning it may not make much sense since we have a callback triggered when the scan is done, right?

Yes, SCANNING->SCAN_DONE->SCAN_RESULTS fetch, supplicant FSM is no impacted in these actions.

Reply:
Ok, thanks for the clarification.